Feature/zone traffic drawers

johncowen opened the pull request:
Whilst there looks like there is a lot going on here, its mainly copy/pasted from the similar view over in Dataplanes and most of the code here was pre-existing.

---

This PR replicates the Dataplane Traffic drawers from over in the dataplanes section into both ZoneIngress and ZoneEgress.

I began by trying to make the code used for Dataplanes more reusable so I coulduse it for both ZoneIngress and ZoneEgress. I found that for several reasons this was overly complicated to do and would have required quite a lot of change within Dataplanes also.

For instance:

- explicitly named route params such as `params.dataPlane`, `params.zoneIngress` and `params.zoneEgress`
- explicitly named route props such as `props.zoneIngress` and `props.zoneEgress`
- explicitly named source URIs `zone-ingress`/`zone-egress` (this one was relatively easy to address and is addressed in this PR)
- Dataplane Proxy URIs require a `mesh` specifying whereas Zone Proxy URIs don't
- ...probably more edge cases

Therefore I chose to mostly copy/paste over the approach from Dataplanes to ZoneIngress, and then again to ZoneEgress (ZoneIngress and ZoneEgress modules are mostly copy pasted anyway).

I plan to make some follow up PRs to reduce the repetition here.

I'll try and add some inlines, but please ask questions if anything isn't clear.

johncowen commented on packages/kuma-gui/src/app/zone-egresses/data/index.ts:
```diff
@@ -8,39 +8,21 @@ import type {
+      networking: {
+        ...item.networking,
+        inboundAddress: item.networking?.address && item.networking?.port ? `${item.networking.address}:${item.networking.port}` : '',
+      },
```
I myself have questions over `socketAddress` and `inboundAddress`. Zones already had `socketAddress` but Dataplanes don't. Dataplanes use `networking.inboundAddress` for this feature, so I copied that over here. In a later PR I may change this slightly and try to organize it so it works for all "proxy types" (all proxy types being ingress, egress _and_ dataplane)

johncowen commented on packages/kuma-gui/src/app/zone-egresses/routes.ts:
```diff
@@ -14,7 +14,59 @@ export const routes = (prefix = 'egresses') => {
+                  },
+
+                ]
+              })('zone-egress'),
```
Perhaps temporarily _not_ using the `connections` module routes and view.vue files.

johncowen reviewed (commented):
Notes:

schogges commented on packages/kuma-gui/src/app/zone-ingresses/views/ConnectionInboundSummaryClustersView.vue:
```diff
@@ -0,0 +1,66 @@
+                  appearance="primary"
+                  @click="refresh"
+                >
+                  Refresh
```
Another i18n `common.refresh`? 🙂

schogges commented on packages/kuma-gui/src/app/zone-egresses/views/ConnectionOutboundSummaryView.vue:
```diff
@@ -0,0 +1,63 @@
+    <AppView>
+      <template #title>
+        <h2>
+          Outbound {{ route.params.connection }}
```
Should we add this to i18n?

schogges commented on packages/kuma-gui/src/app/zone-ingresses/views/ConnectionInboundSummaryStatsView.vue:
```diff
@@ -0,0 +1,68 @@
+                appearance="primary"
+                @click="refresh"
+              >
+                Refresh
```
Another i18n `common.refresh`?

schogges commented on packages/kuma-gui/src/app/zone-ingresses/views/ConnectionOutboundSummaryClustersView.vue:
```diff
@@ -0,0 +1,66 @@
+                  appearance="primary"
+                  @click="refresh"
+                >
+                  Refresh
```
Another i18n `common.refresh`?

schogges commented on packages/kuma-gui/src/app/zone-ingresses/views/ConnectionOutboundSummaryStatsView.vue:
```diff
@@ -0,0 +1,65 @@
+                appearance="primary"
+                @click="refresh"
+              >
+                Refresh
```
Another i18n `common.refresh`?

schogges reviewed (commented):
Nice! LGTM 👍 
just a few nits and minor things 🙂

johncowen commented on packages/kuma-gui/src/app/zone-egresses/views/ConnectionInboundSummaryStatsView.vue:
```diff
@@ -0,0 +1,68 @@
+        v-slot="{ data: stats, refresh }"
+      >
+        <DataCollection
+          :items="stats!.raw.split('\n')"
```
Thanks for checking! Sounds like this is no longer needed, but as its copy/paste the very next PR after this one will be removing this in Zones and where I copied it from (Dataplanes), just trying to stay in scope of zones here.

johncowen commented on packages/kuma-gui/src/app/zone-egresses/views/ConnectionInboundSummaryStatsView.vue:
```diff
@@ -0,0 +1,68 @@
+                appearance="primary"
+                @click="refresh"
+              >
+                Refresh
```
We are gonna circle back on these i18n things, will resolve all these for now

johncowen commented:
Ok I've added a commit which begins to address some of the complications around duplication, the extra commit specifically addresses the first point in the PR description:

> - explicitly named route params such as params.dataPlane, params.zoneIngress and params.zoneEgress

We went through offline, but TLDR is this commit changes and `route.params.zoneIngress`/`route.params.zoneEgress` to instead use `route.params.proxy` which is generic enough to refer to both type of zone proxy.

There was a little extra plumbing required to move it to use `.proxy` but we've moved one step forwards in that we no have a set of 'connections' view files that are shared between ZoneIngress/ZoneEgress.

I'm going to continue working, but in my mind this PR is still fine to go in. If its still here when I get to the next stopping point, I can either add a commit on the end if not I'll just make a new PR.

Apologies that this PR takes a bit of a haphazard approach from a PR/commit perspective, I'm trying to keep the PR scoped, as small as possible but still provide the backstory to the changes being made!

schogges commented on packages/kuma-gui/src/app/connections/views/zones/ConnectionInboundSummaryClustersView.vue:
```diff
@@ -0,0 +1,67 @@
+                  appearance="primary"
+                  @click="refresh"
+                >
+                  Refresh
```
Could we use i18n `common.refresh`? 🙂

johncowen replied:
Honestly I would rather do these as a "lets i18n some remaining text" PRs. If I do them here, then we are likely to miss other places that this has been duplicated from, or just do it badly without taking other things into account.

If we had to i18n the entire app, it wouldn't take us long to do, and I believe there are things that would actually prevent us from having the entire app properly i18n'ed (text in other dependencies that we don't control)

So for me it still stands that I want to stop digging little holes in all the wrong places, and potentially making wide ranging mistakes.

We are a little further along than we were now, we have `<XI18n />` which also potentially gives us prefixed keys. But I'd still prefer to wait a little longer before we do some "lets i18n some remaining text" PRs.

If it comes to it that we need to have our app i18n'ed asap, then I'd be happy to go through and do the remaining texts right now, but I don't think we are at that stage yet.

johncowen replied:
Let me know on this one, I'd be happy to change this bit of text if you felt strongly about it, but all in all I'd prefer to do a piece of work specifically scoped to https://github.com/kumahq/kuma-gui/issues/2813 (having thought about https://github.com/kumahq/kuma-gui/issues/2814 first)

schogges replied:
Fair points, I'm fine either way 🙂

schogges reviewed (approved):
LGTM and I really like the deduplication 👍 🙂 
There are still some of the i18n things, but I'll approve anyways to unblock this since these are just nits 🙂
